ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/AppendOnlyCommandDeduplicationParallelIT.scala

```diff
@@ -63,33 +59,44 @@ class AppendOnlyCommandDeduplicationParallelIT extends LedgerTestSuite {
     allocate(SingleParty),
     runConcurrently = false,
   )(implicit ec => { case Participants(Participant(ledger, party)) =>
-    val deduplicationDuration = 3.seconds
-    val request = ledger
-      .submitAndWaitRequest(party, Dummy(party).create.command)
-      .update(
-        _.commands.deduplicationDuration := deduplicationDuration.asProtobuf
-      )
+    val request = submitAndWaitRequest(ledger, party)
+    runTestWithSubmission[SubmitAndWaitRequest](
+      ledger,
+      party,
+      () => submitAndWaitRequestAndGetStatusCode(ledger)(request, party),
+    )
+  })
+
+  test(
+    s"DeduplicateParallelSubmissionsUsingMixedCommandServiceAndCommandSubmissionService",
+    "Commands submitted at the same, in parallel, using the CommandService the CommandSubmissionService, should be deduplicated",
+    allocate(SingleParty),
+    runConcurrently = false,
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    val submitAndWaitRequest = submitAndWaitRequest(ledger, party)
+    val submitRequest = submitRequest(ledger, party)
     runTestWithSubmission[SubmitAndWaitRequest](
       ledger,
       party,
-      request,
-      submitAndWaitRequestAndGetStatusCode(ledger)(_, party),
+      () =>
+        if (Random.nextBoolean())
+          submitAndWaitRequestAndGetStatusCode(ledger)(submitAndWaitRequest, party)
+        else submitRequestAndGetStatusCode(ledger)(submitRequest, party),
     )
   })
 
   private def runTestWithSubmission[T](
       ledger: ParticipantTestContext,
       party: Party,
-      request: T,
-      submitRequestAndGetStatus: T => Future[Code],
+      submitRequestAndGetStatus: () => Future[Code],
   )(implicit
       ec: ExecutionContext
   ) = {
     val numberOfParallelRequests = 10
```

Comment:
I think it's clearer if this knob controlled by the caller or is at least a suite-scoped constant.

Reply:
I extracted it into the suite scope